File: Lab1/PythonImplementation/fileIO.py
import os

def read_raw_string(filename):
    try:
        with open(filename, 'r') as f:
            content = f.read().strip()
            if not content:
                raise ValueError(f"{filename} is empty")
            return content
    except FileNotFoundError:
        print(f"CRITICAL ERROR: The file '{filename}' was not found.")
        return None
    except Exception as e:
        print(f"SYSTEM ERROR: Could not read {filename}: {e}")
        return None
def write_raw_string(filename, data):
    try:
        with open(filename, 'w') as f:
            f.write(data)
    except PermissionError:
        print(f"CRITICAL ERROR: Permission denied writing to {filename}.")
        
def write_raw_string(filename, data):
    with open(filename, 'w') as f:
        f.write(data)

def write_encrypted_blocks(filename, blocks):
    with open(filename, 'w') as f:
        # Saving as comma-separated integers to avoid encoding issues
        f.write(",".join(map(str, blocks)))

def read_encrypted_blocks(filename):
    if not os.path.exists(filename):
        return []
    with open(filename, 'r') as f:
        content = f.read().strip()
        return [int(b) for b in content.split(",")] if content else []

File: SymmetricEncryptionSimpleDES/RustImplementation/rust_encryption/src/file_io.rs
use std::fs;

pub fn read_raw_string(filename: &str) -> Option<String> {
    fs::read_to_string(filename).ok()
}

pub fn write_raw_string(filename: &str, data: &str) {
    let _ = fs::write(filename, data);
}

pub fn write_encrypted_blocks(filename: &str, blocks: &[u16]) {
    let content = blocks
        .iter()
        .map(|b| b.to_string())
        .collect::<Vec<_>>()
        .join(",");
    let _ = fs::write(filename, content);
}

pub fn read_encrypted_blocks(filename: &str) -> Vec<u16> {
    fs::read_to_string(filename)
        .unwrap_or_default()
        .split(',')
        .filter_map(|s| s.parse::<u16>().ok())
        .collect()
}